File: Lecture 6/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/MoviesAdapter.java
package com.inovaeg.androidtrainingpharos2020;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MoviesAdapter extends RecyclerView.Adapter<MovieViewHolder> {
    private ArrayList<Movie> movies;
    private final int TYPE_HEADER = 0;
    private final int TYPE_MOVIE = 1;

    public MoviesAdapter(ArrayList<Movie> movies) {
        this.movies = movies;
    }

    @NonNull
    @Override
    public MovieViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        int cellId;
        if (viewType == TYPE_HEADER) {
            cellId = R.layout.cell_header;
        } else {
            cellId = R.layout.cell_movie;
        }
        View view = LayoutInflater.from(parent.getContext()).inflate(cellId, parent, false);
        return new MovieViewHolder(view, parent.getContext());
    }

    @Override
    public void onBindViewHolder(@NonNull MovieViewHolder holder, int position) {
        holder.bind(movies.get(position));
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }

    public void appendData(ArrayList<Movie> newData) {
        movies.addAll(newData);
        notifyDataSetChanged();
    }

    @Override
    public int getItemViewType(int position) {
//        if (position == 0)
//            return TYPE_HEADER;
//        else
        return TYPE_MOVIE;
    }
}

File: Lecture 5/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/MoviesAdapter.kt
package com.inovaeg.androidtrainingpharos2020

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class MoviesAdapter(private val movies: ArrayList<Movie>) : RecyclerView.Adapter<MovieViewHolder>() {
    private val TYPE_HEADER = 0
    private val TYPE_MOVIE = 1
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
        val cellId: Int
        cellId = if (viewType == TYPE_HEADER) {
            R.layout.cell_header
        } else {
            R.layout.cell_movie
        }
        val view = LayoutInflater.from(parent.context).inflate(cellId, parent, false)
        return MovieViewHolder(view, parent.context)
    }

    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
        holder.bind(movies[position])
    }

    override fun getItemCount(): Int {
        return movies.size
    }

    fun appendDate(newData: ArrayList<Movie>) {
        movies.addAll(newData)
        notifyDataSetChanged()
    }

    override fun getItemViewType(position: Int): Int {
        return if (position == 0) TYPE_HEADER else TYPE_MOVIE
    }

}